### PR TITLE
bsdlib: version 0.7.3

### PR DESCRIPTION
**Pull request by kpuusaari**
Library version: fe95416e00492997f4f4fb3170a4cfb49781e17f

- Aligned the naming of nrf_pollfd structure elements with pollfd
- Fixed IP socket state after accept() call

Signed-off-by: Kimmo Puusaari <kimmo.puusaari@nordicsemi.no>

### CODE REVIEW

**Comment by rlubos** [comment]
@thst-nordic For some reason, this PR is treated as an external contribution, please have a look.

**Comment by eibe85** [comment]
@rlubos perhaps due to the change in remote urls? 
@kpuusaari is there a coresponding nrf PR to test this with, with a manifest update?

**Comment by kpuusaari** [comment]
> 
> 
> @rlubos perhaps due to the change in remote urls?
> @kpuusaari is there a coresponding nrf PR to test this with, with a manifest update?

https://github.com/nrfconnect/sdk-nrf/pull/2307

**Comment by rlubos** [comment]
@b-gent Please have a look at the changelog so we can merge this.

File: bsdlib/include/nrf_socket.h [hunk 1]
```diff
@@ -1264,7 +1300,7 @@ int nrf_select(int                        nfds,
  * @brief Method to poll for events on one or more sockets.
  *
  * @param[inout] An array of sockets, and respective for each socket that the caller polls for.
```
Comment (b-gent) [inline bsdlib/include/nrf_socket.h:1302]:
```suggestion
 * @param[in,out] An array of sockets, and respective for each socket that the caller polls for.
```

File: bsdlib/include/nrf_socket.h [hunk 2]
```diff
@@ -1264,7 +1300,7 @@ int nrf_select(int                        nfds,
...
- *               The occurred events per socket is returned in the requested field of @ref struct nrf_pollfd structure.
+ *               The occurred events per socket is returned in the revents field of @ref struct nrf_pollfd structure.
  *               Shall not be NULL.
  * @param[in]    Positive number of sockets being polled for events.
```
Comment (b-gent) [inline bsdlib/include/nrf_socket.h:1305]:
```suggestion
 * @param[in] nfds    Positive number of sockets being polled for events.
```

Also below ( I can't comment there as it's outside of PR scope):
1. In line 1308: ``@param[in] timeout   Timeout in milliseconds.``
2. In line 1311: ``@return A positive number less than or equal to nfds indicating sockets on which events occurred.``

File: bsdlib/include/nrf_socket.h [hunk 3]
```diff
@@ -1243,9 +1279,9 @@ int nrf_select(int                        nfds,
  */
  struct nrf_pollfd
  {
-    int   handle;        /**< Socket handle */
-    short requested;     /**< Requested events, is a mask of events */
-    short returned;      /**< Returned events, is a mask of events */
+    int   fd;         /**< Socket handle */
+    short events;     /**< Requested events, is a mask of events */
+    short revents;    /**< Returned  events, is a mask of events */
```
Comment (b-gent) [inline bsdlib/include/nrf_socket.h:1284]: - double space in this line, I think
- please end each description with a dot

File: bsdlib/include/nrf_socket.h [hunk 4]
```diff
@@ -1003,6 +1018,27 @@ typedef uint16_t nrf_gnss_nmea_mask_t;
  *          - @c NRF_GNSS_PSM_DUTY_CYCLING_PERFORMANCE for low power mode with better performance.
  *          - @c NRF_GNSS_PSM_DUTY_CYCLING_POWER for low power mode with lower power consumption.
  *
+ * The GNSS receiver may track signals (i.e. run) either continuously or in duty
```
Comment (b-gent) [inline bsdlib/include/nrf_socket.h:1021]: This information is now here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrfxlib/bsdlib/doc/gnss_extension.html

We discussed this in the previous PR , please do not put conceptual documentation into API documentation.

File: bsdlib/include/nrf_socket.h [hunk 5]
```diff
@@ -275,12 +275,27 @@ typedef uint32_t nrf_fd_set;
...
  *
  * @details
+ * Fix interval parameter controls the time between GNSS receiver starts.
+ * It also controls whether the GNSS receiver is stopped after a valid PVT
+ * estimate. The fix interval parameter determines the mode of navigation.
+ * There are three different navigation modes available: single-fix,
+ * continuous, and periodic.
+ *
  * Single-fix navigation mode is engaged by setting the fix interval to 0.
+ * In this mode once a valid PVT estimate is produced, the GNSS receiver is
```
Comment (b-gent) [inline bsdlib/include/nrf_socket.h:285]: Please do not describe the modes here, this information is in https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrfxlib/bsdlib/doc/gnss_extension.html

File: bsdlib/doc/CHANGELOG.rst [hunk 6]
```diff
@@ -5,6 +5,12 @@ Changelog
 
 All notable changes to this project are documented in this file.
 
+bsdlib 0.7.3
+************
+
+* Aligned the naming of ``nrf_pollfd`` structure elements with ``pollfd``
```
Comment (b-gent) [inline bsdlib/doc/CHANGELOG.rst:11]:
```suggestion
* Aligned the naming of ``nrf_pollfd`` structure elements with ``pollfd``.
```

File: bsdlib/doc/CHANGELOG.rst [hunk 7]
```diff
@@ -5,6 +5,12 @@ Changelog
...
+* Fixed IP socket state after :cpp:func:`accept()` call
```
Comment (b-gent) [inline bsdlib/doc/CHANGELOG.rst:12]:
```suggestion
* Fixed IP socket state after :cpp:func:`accept()` call.
```

File: bsdlib/include/nrf_socket.h (hunk 4 above)
Comment (kpuusaari) [reply bsdlib/include/nrf_socket.h:1021]: yes, there was an earlier backport issue, will do fixes and take your suggestions to that. Thanks!

File: bsdlib/include/nrf_socket.h (hunk 1 above)
Comment (kpuusaari) [reply bsdlib/include/nrf_socket.h:1302]: ok, changing also one inout to in,out later in file (p_optval)

File: bsdlib/include/nrf_socket.h (hunk 2 above)
Comment (kpuusaari) [reply bsdlib/include/nrf_socket.h:1305]: ok

File: bsdlib/include/nrf_socket.h (hunk 5 above)
Comment (kpuusaari) [reply bsdlib/include/nrf_socket.h:285]: same

File: bsdlib/doc/CHANGELOG.rst (hunk 6 above)
Comment (kpuusaari) [reply bsdlib/doc/CHANGELOG.rst:11]: ok

File: bsdlib/doc/CHANGELOG.rst (hunk 7 above)
Comment (kpuusaari) [reply bsdlib/doc/CHANGELOG.rst:12]: ok

**Pull request merged**